Mathlib/RingTheory/Ideal/Maps.lean

```diff
@@ -385,6 +398,51 @@ theorem mem_map_of_equiv {E : Type*} [EquivLike E R S] [RingEquivClass E R S] (e
   · rintro ⟨x, hx, rfl⟩
     exact mem_map_of_mem e hx
 
+section Bijective
+
+/-- Special case of the correspondence theorem for isomorphic rings -/
+def relIsoOfBijective (hf : Function.Bijective f) : Ideal S ≃o Ideal R where
+  toFun := comap f
+  invFun := map f
+  left_inv := map_comap_of_surjective _ hf.2
+  right_inv J := le_antisymm (fun _ h ↦
+    have ⟨y, hy, eq⟩ := (mem_map_iff_of_surjective _ hf.2).mp h; hf.1 eq ▸ hy) le_comap_map
+  map_rel_iff' {_ _} := by
+    refine ⟨fun h ↦ ?_, comap_mono⟩
+    have := map_mono (f := f) h
+    simpa only [Equiv.coe_fn_mk, map_comap_of_surjective f hf.2] using this
+
+theorem comap_le_iff_le_map (hf : Function.Bijective f) {I : Ideal R} {K : Ideal S} :
+    comap f K ≤ I ↔ K ≤ map f I :=
+  ⟨fun h => le_map_of_comap_le_of_surjective f hf.right h, fun h =>
+    (relIsoOfBijective f hf).right_inv I ▸ comap_mono h⟩
+
+lemma comap_map_of_bijective (hf : Function.Bijective f) {I : Ideal R} :
+    (I.map f).comap f = I :=
+  le_antisymm ((comap_le_iff_le_map f hf).mpr fun _ ↦ id) le_comap_map
+
+theorem map.isMaximal (hf : Function.Bijective f) {I : Ideal R} (H : IsMaximal I) :
```

Comment:
I think `IsMaximal.map` would make more sense and allow dot notation. Same for comap